usersync/syncersbuilder_test.go

```diff
@@ -330,3 +330,88 @@ func TestChooseSyncerConfig(t *testing.T) {
 		}
 	}
 }
+
+func TestGetSyncerKey(t *testing.T) {
+	redirectCfg := config.SyncerEndpoint{URL: "redirect-url", UserMacro: "$UID"}
+	syncer1 := config.Syncer{Key: "", Redirect: &redirectCfg}
+	syncer2 := syncer1
+	syncer3 := config.Syncer{Key: "key-1", Redirect: &redirectCfg}
+	syncer4 := config.Syncer{Key: "key-1"}
+
+	biddersWithSyncerCfg := config.BidderInfos{
+		"parent-1": config.BidderInfo{Syncer: &syncer1},
+		"alias-1":  config.BidderInfo{AliasOf: "parent-1", Syncer: &syncer1},
+		"alias-2":  config.BidderInfo{AliasOf: "parent-1", Syncer: &syncer2},
+		"parent-2": config.BidderInfo{Syncer: &syncer3},
+		"alias-3":  config.BidderInfo{AliasOf: "parent-1", Syncer: &syncer3},
+		"alias-4":  config.BidderInfo{AliasOf: "parent-2", Syncer: &syncer4},
+	}
+
+	tests := []struct {
+		name                 string
+		biddersWithSyncerCfg map[string]config.BidderInfo
+		bidderName           string
+		bidderInfo           config.BidderInfo
+		expectedErr          string
+		expectedSyncerKey    string
+	}{
+		{
+			name:        "alias_bidder_has_no_syncer_config",
+			bidderName:  "alias-1",
+			bidderInfo:  config.BidderInfo{Syncer: nil},
+			expectedErr: "found no syncer config for alias bidder alias-1",
+		},
```

Comment:
This test means it is mandatory for any bidder and not just an alias to define userSync and that might not be true.

Reply:
This test is for `getSyncerKey()` . So it is mandatory for `bidderInfo`  (passed as input to getSyncerKey()) to have `userSync` details for determining syncer key.  

https://github.com/prebid/prebid-server/blob/eb836cd4ae49b985afee36a0bd030fddd6e3edc7/usersync/syncersbuilder.go#L156-L159


Refer usage of `getSyncerKey()`. We have `BuildSyncer()` that calls `getSyncerKey()` only for bidders who has syncer config.  So `BuildSyncer()` will never pass  `bidderInfo`  without syncer config. 

But from standalone function point of view,  `getSyncerKey()` should check if bidder has syncer config. Else function can decide syncer key 
https://github.com/prebid/prebid-server/blob/eb836cd4ae49b985afee36a0bd030fddd6e3edc7/usersync/syncersbuilder.go#L38-L47